### PR TITLE
Update repo structure 0.5 #211

### LINKED ISSUE #211: Provide repository overview document
https://github.com/openebs/ has multiple active repositories. While https://github.com/openebs/openebs is the starting point, the contributors will need information on:
- Where they can find the implementation of different modules? (maya/jiva)
- Where should they raise new issues ?

### PR DESCRIPTION
**Pull request by kmova**
**What this PR does / why we need it**: Update the overview document with 0.5 design and repository information. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #211 

**Special notes for your reviewer**:

### CODE REVIEW

File: contribute/design/README.md [hunk 1]
```diff
@@ -1,29 +1,58 @@
...
+Yes! The storage controller functionality is itself, delivered as containers. An OpenEBS Volume comprises of one or more containers working as a clustered microservice, providing block storage to other containers. This micro-services based architecture for storage controller functionality - where each Volume data is being served by its own set of containers, as opposed to a single monolothic storage controller software serving multiples volumes is what differentiates OpenEBS from traditional storage appliances.
```
Comment (mahebbar) [inline contribute/design/README.md]: where each volume data is served by its own set of containers as against a single monolithic storage controller serving multiple volumes. This is what differentiates OpenEBS from traditional storage appliances.

File: contribute/design/README.md [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
 
-OpenEBS, being container native, seamlessely integrates easily into the container orchestartors via OpenEBS Maya and OpenEBS Volume containers. 
+OpenEBS Storage Controller functionality comprises of serveral micro-services (or containers) that can be classified into two broad categories: OpenEBS Data Plane - that serves the data to the applications and OpenEBS Control Plane - that manages the OpenEBS Volume Containers. If you notice this classification, closely resembling that of a typical Container Orchestrator, it is because OpenEBS Volumes are delivered through containers and these containers are better managed by Container Orchestrators! So OpenEBS Control Plane services work in conjunction with the CO - schedulers, apiserver, etc.,
```
Comment (mahebbar) [inline contribute/design/README.md]: ... into two broad categories:
*  OpenEBS Data Plane - ...
* OpenEBS Control Plane - ...
closely resembling that of a typical = resembling a typical
So OpenEBS Control Plane = Hence, OpenEBS Control Plane 
CO - to expand this?
etc. = and so on.

File: contribute/design/README.md [hunk 3]
```diff
@@ -1,29 +1,58 @@
...
+OpenEBS Storage Controller functionality comprises of serveral micro-services (or containers) that can be classified into two broad categories: OpenEBS Data Plane - that serves the data to the applications and OpenEBS Control Plane - that manages the OpenEBS Volume Containers. If you notice this classification, closely resembling that of a typical Container Orchestrator, it is because OpenEBS Volumes are delivered through containers and these containers are better managed by Container Orchestrators! So OpenEBS Control Plane services work in conjunction with the CO - schedulers, apiserver, etc., 
+ 
+## OpenEBS Volume Container (aka jiva, aka data plane)
 
-## OpenEBS Volume Containers (aka jiva)
+*openebs/jiva* containers are at the heart of the OpenEBS Volume's storage functionality. OpenEBS Volumes are fully isolated user space storage engines that present the block storage via iSCSI and persist the data onto the storage attached to Nodes (aka Docker Hosts). The storage attached to the docker hosts can either be directly attached disks or cloud volumes (like EBS, GPD, etc.,) depending on where the Container Cluster is deployed.
```
Comment (mahebbar) [inline contribute/design/README.md]: etc. = and so on

File: contribute/design/README.md [hunk 4]
```diff
@@ -1,29 +1,58 @@
...
 
-*openebs/jiva* containers are the heart of the OpenEBS Volume's storage functionality. OpenEBS Volumes provide persistent storage for containers, with resliency against system failures via a distrubuted architecture, faster access to the storage, snapshot and backup capabilities. In addition, provides mechanism for monitoring the usage and enforcing QoS policies. 
+![OpenEBS Data Plane](../../documentation/source/_static/OpenEBS-Data-Plane.png)
 
-OpenEBS Volume (or Jiva) Containes are scheduled onto the container hosts using Orchestration Engines. Jiva containers consume the disk storage from the local host or remote storage using volume plugins (like k8s persistent volumes). OpenEBS Volumes are described in yaml files, just like your application pods, are deployed through objects like Pods, Deployments, Services, StatefulSets etc., 
+OpenEBS Volumes provide persistent storage for containers, with resliency against system failures, faster access to the storage, snapshot and backup capabilities. In addition, provides mechanism for monitoring the usage and enforcing QoS policies.
```
Comment (mahebbar) [inline contribute/design/README.md]: resliency = resilency
In addition,  provides = in addition, it provides a

File: contribute/design/README.md [hunk 5]
```diff
@@ -1,29 +1,58 @@
...
 
-OpenEBS Volume (or Jiva) Containes are scheduled onto the container hosts using Orchestration Engines. Jiva containers consume the disk storage from the local host or remote storage using volume plugins (like k8s persistent volumes). OpenEBS Volumes are described in yaml files, just like your application pods, are deployed through objects like Pods, Deployments, Services, StatefulSets etc., 
+OpenEBS Volumes provide persistent storage for containers, with resliency against system failures, faster access to the storage, snapshot and backup capabilities. In addition, provides mechanism for monitoring the usage and enforcing QoS policies. 
 
-Further details on the OpenEBS Volume Container - jiva can be found under the following repo - [openeb/jiva](https://github.com/openebs/jiva)
+The Disks where the data is peristed are called as *Storage Backends*, which can be either host directories, attached block devices or remote disks. Each OpenEBS Volume comprises of an iSCSI Target Container (represented as openebs-vol1 in the above diagram) and one or more Replica Containers (openebs-vol1-R1 and openebs-vol1-R2).
```
Comment (mahebbar) [inline contribute/design/README.md]: The Disks where data is persisted
peristed = persisted
which can be either = which can either be 
devices = devices,

File: contribute/design/README.md [hunk 6]
```diff
@@ -1,29 +1,58 @@
...
 
-## OpenEBS Storage Orchestration (aka maya)
+The application pods will access the storage via the iSCSI Target Container, whicn will replicate the data to all its replica's. In the event of the node failure, the iSCSI Target Container is started on one of the remaining online nodes, and will serve the data by connecting to the available Replica container(s).
```
Comment (mahebbar) [inline contribute/design/README.md]: will access = access the
whicn = which
will relicate = replicates
the node failure = node failure
Target Container is started = Target Container starts on one of the...
container(s) = containers

File: contribute/design/README.md [hunk 7]
```diff
@@ -1,29 +1,58 @@
...
+### Source Code
 
-Maya is a set of components that can be further divided as follows:
-- Services : These encapsulate a specific functionality like maya-apiserver, maya-agent, maya-mulebot, etc., that are either installed as binary services or container services. These will be either be running as Cluster Services (like the maya-apiserver), or on every docker host or node (like maya-agent) or can be completely out-of-band like (storage-analytics in the cloud). You can learn more about these services under the following repo - [openebs/maya](https://github.com/openebs/maya).
+- [openebs/jiva](https://github.com/openebs/jiva) Storage Controller functionality which includes the replication logic
```
Comment (mahebbar) [inline contribute/design/README.md]: remove which

File: contribute/design/README.md [hunk 8]
```diff
@@ -1,29 +1,58 @@
...
+### Source Code
 
-Maya is a set of components that can be further divided as follows:
-- Services : These encapsulate a specific functionality like maya-apiserver, maya-agent, maya-mulebot, etc., that are either installed as binary services or container services. These will be either be running as Cluster Services (like the maya-apiserver), or on every docker host or node (like maya-agent) or can be completely out-of-band like (storage-analytics in the cloud). You can learn more about these services under the following repo - [openebs/maya](https://github.com/openebs/maya).
+- [openebs/jiva](https://github.com/openebs/jiva) Storage Controller functionality which includes the replication logic 
+- [openebs/gotgt](https://github.com/openebs/gotgt) iSCSI Target functionality, which is used by *openebs/jiva*.
```
Comment (mahebbar) [inline contribute/design/README.md]: remove which

File: contribute/design/README.md [hunk 9]
```diff
@@ -1,29 +1,58 @@
...
 
-- Plugins : Maya allows the OpenEBS storage to be run along side any orchestration engine like kubernetes, docker swarm, nomad etc., This is made possible by the maya orchestration engine plugins. For example, one of the ways in which Kubernetes can consume OpenEBS Storage is via the K8s Dynamic Provisioners - [openebs-provisioner](https://github.com/openebs/external-storage/tree/master/openebs). 
 
-- Tools : CLI/UI tools that will help with installation or management of tasks. One of the first tools under heavy development is [mayactl](https://github.com/openebs/maya/tree/master/cmd/mayactl), that aims at simplying the management of OpenEBS Volumes. 
+
+## OpenEBS Control Plane (aka OpenEBS Storage Orchestration, aka maya)
+
+OpenEBS Control Plane - auguments the functionality provided by the Container Orchestrator, with storage specific orchestration capabilities, via a set of services. OpenEBS control plane can hook into any of the container orchestrators like Kubernetes, Docker Swarm, Nomad, etc., making it possible to run OpenEBS hyper-coverged with Container Orchestrators.
```
Comment (mahebbar) [inline contribute/design/README.md]: hyper-coverged = hyper-converged

File: contribute/design/README.md [hunk 10]
```diff
@@ -1,29 +1,58 @@
...
+
+OpenEBS Control Plane - auguments the functionality provided by the Container Orchestrator, with storage specific orchestration capabilities, via a set of services. OpenEBS control plane can hook into any of the container orchestrators like Kubernetes, Docker Swarm, Nomad, etc., making it possible to run OpenEBS hyper-coverged with Container Orchestrators. 
+
+![OpenEBS Control Plane](../../documentation/source/_static/OpenEBS-Control-Plane.png)
+
+OpenEBS Control Plane is also delivered as micro-services, where the services built can be classified as follows:
+- Plugins into Container Orchestrotors, which augument the functionality by hooking into the framework provided like:
+  - K8s Dynamic Provisioners - [openebs-provisioner](https://github.com/openebs/external-storage/tree/master/openebs)
+  - Kube Dashboard - [openebs-dashboard](https://github.com/openebs/dashboard)
+  - Extended Schema - like CRDs in case of Kubernetes to storage the OpenEBS specific configuration related data
```
Comment (mahebbar) [inline contribute/design/README.md]: storage = store

File: contribute/design/README.md [hunk 11]
```diff
@@ -1,29 +1,58 @@
...
+- Cluster Services, which provide OpenEBS specific storage intelligence like:
```
Comment (mahebbar) [inline contribute/design/README.md]: Cluster Services - provides

File: contribute/design/README.md [hunk 12]
```diff
@@ -1,29 +1,58 @@
...
+  - maya-apiserver - includes the API for performing Volume operations that can translate requests nto CO specific operations
```
Comment (mahebbar) [inline contribute/design/README.md]: nto - into

File: contribute/design/README.md [hunk 13]
```diff
@@ -1,29 +1,58 @@
...
+  - maya-mulebot - uses the information gathered to suggest optimized placement and event handling tips
+  - maya-connect - allows to upload the monitoring data to maya-cloud for further storage analytics
```
Comment (mahebbar) [inline contribute/design/README.md]: monitoring data = monitored data 
storage analytics = storage analysis?

File: contribute/design/README.md [hunk 14]
```diff
@@ -1,29 +1,58 @@
...
+- Node Services, which provide OpenEBS specific storage intelligence, but run along-side kubelet like:
```
Comment (mahebbar) [inline contribute/design/README.md]: Node Services - provides

File: contribute/design/README.md [hunk 15]
```diff
@@ -1,29 +1,58 @@
...
+  - maya-agent - includes storage management functionality 
+
+Monitoring and Tracing capabilities are added by instrumenting the above services with prometheus, heapster, grafana and jaegar. 
+
+*Note: In case of K8s, the monitoring aspects like heapster can be changed in future, depending on the outcome from sig-instrumention design proposals*
+
+### Source Code
+
+- [openebs/maya](https://github.com/openebs/maya) The code for all the binaries that are specific (non-plugins) are stored in this repository - like maya-apiserver, maya-agent, maya-mulebot, maya-connect, mayactl.
```
Comment (mahebbar) [inline contribute/design/README.md]: for all binaries
and, mayactl

File: contribute/design/README.md [hunk 16]
```diff
@@ -1,29 +1,58 @@
...
+  - maya-agent - includes storage management functionality 
+
+Monitoring and Tracing capabilities are added by instrumenting the above services with prometheus, heapster, grafana and jaegar. 
+
+*Note: In case of K8s, the monitoring aspects like heapster can be changed in future, depending on the outcome from sig-instrumention design proposals*
+
+### Source Code
+
+- [openebs/maya](https://github.com/openebs/maya) The code for all the binaries that are specific (non-plugins) are stored in this repository - like maya-apiserver, maya-agent, maya-mulebot, maya-connect, mayactl. 
+- [openebs-dashboard](https://github.com/openebs/dashboard) A fork of the kubenetes dashboard project, that is extended with storage functionality.
```
Comment (mahebbar) [inline contribute/design/README.md]: kubernetes = Kubernetes

File: contribute/design/README.md [hunk 17]
```diff
@@ -1,29 +1,58 @@
...
+ 
+## OpenEBS Volume Container (aka jiva, aka data plane)
 
-## OpenEBS Volume Containers (aka jiva)
+*openebs/jiva* containers are at the heart of the OpenEBS Volume's storage functionality. OpenEBS Volumes are fully isolated user space storage engines that present the block storage via iSCSI and persist the data onto the storage attached to Nodes (aka Docker Hosts). The storage attached to the docker hosts can either be directly attached disks or cloud volumes (like EBS, GPD, etc.,) depending on where the Container Cluster is deployed. 
 
-*openebs/jiva* containers are the heart of the OpenEBS Volume's storage functionality. OpenEBS Volumes provide persistent storage for containers, with resliency against system failures via a distrubuted architecture, faster access to the storage, snapshot and backup capabilities. In addition, provides mechanism for monitoring the usage and enforcing QoS policies.
```
Comment (mahebbar) [inline contribute/design/README.md]: resliency = resiliency 
distrubuted - distributed
in addition, it provides a

File: contribute/design/README.md [hunk 18]
```diff
@@ -1,29 +1,58 @@
...
 
-*openebs/jiva* containers are the heart of the OpenEBS Volume's storage functionality. OpenEBS Volumes provide persistent storage for containers, with resliency against system failures via a distrubuted architecture, faster access to the storage, snapshot and backup capabilities. In addition, provides mechanism for monitoring the usage and enforcing QoS policies. 
+![OpenEBS Data Plane](../../documentation/source/_static/OpenEBS-Data-Plane.png)
 
-OpenEBS Volume (or Jiva) Containes are scheduled onto the container hosts using Orchestration Engines. Jiva containers consume the disk storage from the local host or remote storage using volume plugins (like k8s persistent volumes). OpenEBS Volumes are described in yaml files, just like your application pods, are deployed through objects like Pods, Deployments, Services, StatefulSets etc.,
```
Comment (mahebbar) [inline contribute/design/README.md]: Containes = Containers 
, are deployed = , and are deployed
etc. = and so on.

File: contribute/design/README.md [hunk 19]
```diff
@@ -1,29 +1,58 @@
...
+![OpenEBS Data Plane](../../documentation/source/_static/OpenEBS-Data-Plane.png)
 
-OpenEBS Volume (or Jiva) Containes are scheduled onto the container hosts using Orchestration Engines. Jiva containers consume the disk storage from the local host or remote storage using volume plugins (like k8s persistent volumes). OpenEBS Volumes are described in yaml files, just like your application pods, are deployed through objects like Pods, Deployments, Services, StatefulSets etc., 
+OpenEBS Volumes provide persistent storage for containers, with resliency against system failures, faster access to the storage, snapshot and backup capabilities. In addition, provides mechanism for monitoring the usage and enforcing QoS policies. 
 
-Further details on the OpenEBS Volume Container - jiva can be found under the following repo - [openeb/jiva](https://github.com/openebs/jiva)
```
Comment (mahebbar) [inline contribute/design/README.md]: repo = repository

File: contribute/design/README.md [hunk 20]
```diff
@@ -1,29 +1,58 @@
...
 
-Maya makes the storage infrastructure programmable via yaml files that the DevOps can define and commit, just like the container clusters or containerized applications. Maya builds on top of the container orchestration engine capabilities in terms of runtime, scheduling, monitoring etc., and extends the capabilities to orchestrate and simplify the management of storage. Maya will learn and provide storage metrics to the container schedulers for better placements of pods as well as storage migration between hosts within/across cluster(s).
+### Source Code
 
-Maya is a set of components that can be further divided as follows:
-- Services : These encapsulate a specific functionality like maya-apiserver, maya-agent, maya-mulebot, etc., that are either installed as binary services or container services. These will be either be running as Cluster Services (like the maya-apiserver), or on every docker host or node (like maya-agent) or can be completely out-of-band like (storage-analytics in the cloud). You can learn more about these services under the following repo - [openebs/maya](https://github.com/openebs/maya).
```
Comment (mahebbar) [inline contribute/design/README.md]: etc. = and so on
that are either = are either 
These will be = remove be
repo = repository

File: contribute/design/README.md [hunk 21]
```diff
@@ -1,29 +1,58 @@
...
+- [openebs/jiva](https://github.com/openebs/jiva) Storage Controller functionality which includes the replication logic 
+- [openebs/gotgt](https://github.com/openebs/gotgt) iSCSI Target functionality, which is used by *openebs/jiva*.
 
-- Plugins : Maya allows the OpenEBS storage to be run along side any orchestration engine like kubernetes, docker swarm, nomad etc., This is made possible by the maya orchestration engine plugins. For example, one of the ways in which Kubernetes can consume OpenEBS Storage is via the K8s Dynamic Provisioners - [openebs-provisioner](https://github.com/openebs/external-storage/tree/master/openebs).
```
Comment (mahebbar) [inline contribute/design/README.md]: to be run = to run

File: contribute/design/README.md [hunk 22]
```diff
@@ -1,29 +1,58 @@
 # Overview
 
-OpenEBS is a open source storage platform delivering containerized block storage for containers. OpenEBS aims at making storage instrastructure programmable, easy-to-use, consumed by applications via self-service. OpenEBS takes storage software upgrade seriously.
```
Comment (mahebbar) [inline contribute/design/README.md]: instrastructure = infrastructure

File: contribute/design/README.md (hunk 1 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 2 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 3 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 4 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 6 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 7 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 8 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 9 above)
Comment (kmova) [reply contribute/design/README.md]: thanks

File: contribute/design/README.md (hunk 10 above)
Comment (kmova) [reply contribute/design/README.md]: thanks

File: contribute/design/README.md (hunk 12 above)
Comment (kmova) [reply contribute/design/README.md]: done

File: contribute/design/README.md (hunk 17 above)
Comment (kmova) [reply contribute/design/README.md]: deleted line

File: contribute/design/README.md (hunk 18 above)
Comment (kmova) [reply contribute/design/README.md]: delete line

File: contribute/design/README.md [hunk 23]
```diff
@@ -49,8 +52,8 @@ Monitoring and Tracing capabilities are added by instrumenting the above service
 
 ### Source Code
 
-- [openebs/maya](https://github.com/openebs/maya) The code for all the binaries that are specific (non-plugins) are stored in this repository - like maya-apiserver, maya-agent, maya-mulebot, maya-connect, mayactl. 
-- [openebs-dashboard](https://github.com/openebs/dashboard) A fork of the kubenetes dashboard project, that is extended with storage functionality.
+- [openebs/maya](https://github.com/openebs/maya) The code for all binaries that are specific (non-plugins) are stored in this repository - like maya-apiserver, maya-agent, maya-mulebot, maya-connect, mayactl and so on.
+- [openebs-dashboard](https://github.com/openebs/dashboard) A fork of the Kubenetes dashboard project, that is extended with storage functionality.
```
Comment (mahebbar) [inline contribute/design/README.md]: Kubenetes = Kubernetes

**Pull request merged**